build/SourceIndexing/IndexPdbs.ps1

```diff
@@ -0,0 +1,82 @@
+[CmdLetBinding()]
+Param(
+    [Parameter(Mandatory=$true, Position=0)][string]$SearchDir,
+    [Parameter(Mandatory=$true, Position=1)][string]$SourceRoot,
+    [Parameter(Mandatory=$true, Position=2)][string]$CommitId,
+    [string]$Organization = "Microsoft",
+    [string]$Repo = "microsoft-ui-xaml",
+    [switch]$recursive
+)
+
+$debuggerPath = (Get-ItemProperty -path "HKLM:\SOFTWARE\WOW6432Node\Microsoft\Windows Kits\Installed Roots" -name WindowsDebuggersRoot10).WindowsDebuggersRoot10
+$srcsrvPath = Join-Path $debuggerPath "x64\srcsrv"
+$srctoolExe = Join-Path $srcsrvPath "srctool.exe"
+$pdbstrExe = Join-Path $srcsrvPath "pdbstr.exe"
+
+$fileTable = @{}
```

Comment:
>$fileTable [](start = 0, length = 10)

Can we exclude all files related to tests? Also maybe we only need .cpp/.h

Reply:
Indexing test apps and dlls is convenient too because sometimes you want to debug those. Plus we were indexing them before so I wanted to keep parity.

Reply:
> maybe we only need .cpp/.h

Can you elaborate on why you would want to restrict it? Source files could have a variety of extensions and as long as they're a checked-in file in Git that the debugger has source code pointing at then we should map the file... at least that was my thinking.